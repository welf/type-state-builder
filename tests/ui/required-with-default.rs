use type_state_builder::TypeStateBuilder;

#[derive(TypeStateBuilder)]
struct InvalidStruct {
    // This should be an error: required fields cannot have default values
    #[builder(required, default = "String::from(\"default\")")]
    name: String,

    email: Option<String>,
}

fn main() {}
