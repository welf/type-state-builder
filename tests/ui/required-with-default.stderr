error: Required fields cannot have default values. Remove #[builder(default = "...")] or make the field optional by removing #[builder(required)]
 --> tests/ui/required-with-default.rs:7:5
  |
7 |     name: String,
  |     ^^^^

error[E0599]: no function or associated item named `builder` found for struct `InvalidStruct` in the current scope
  --> tests/ui/required-with-default.rs:13:36
   |
4  | struct InvalidStruct {
   | -------------------- function or associated item `builder` not found for this struct
...
13 |     let _instance = InvalidStruct::builder()
   |                                    ^^^^^^^ function or associated item not found in `InvalidStruct`
